### PR TITLE
ListIOCs API to return number of findings per IOC.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Refactored ListIOCs API to return the correct number of findings for each IOC.

### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java [hunk 1]
```diff
@@ -178,12 +188,8 @@ private void listIocs(List<String> iocIndices) {
                 }
                 boolQueryBuilder.must(typeQueryBuilder);
             }
-//             todo remove filter. not needed because feed ids are fetch before listIocs()
-//            if (request.getFeedIds() != null && !request.getFeedIds().isEmpty()) {
-//                boolQueryBuilder.filter(QueryBuilders.termQuery(STIX2_IOC_NESTED_PATH + STIX2IOC.FEED_ID_FIELD, request.getFeedIds()));
-//            }
 
-            if (!request.getTable().getSearchString().isEmpty()) {
+            if (request.getTable().getSearchString() != null && !request.getTable().getSearchString().isEmpty()) {
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java:192]: nit: `StringUtils.isBlank(<str>)`

File: src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java [hunk 2]
```diff
@@ -236,17 +245,66 @@ public void onResponse(SearchResponse searchResponse) {
...
+                                    iocs.add(ioc);
                                 } catch (Exception e) {
                                     log.error(
                                             () -> new ParameterizedMessage("Failed to parse IOC doc from hit {}", hit.getId()), e
                                     );
                                 }
                             });
-                    onOperation(new ListIOCsActionResponse(searchResponse.getHits().getTotalHits().value, iocs));
+
+                    GetIocFindingsRequest getFindingsRequest = new GetIocFindingsRequest(
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java:257]: nit: can we instead do an aggregation query where we get the map of ioc Id to num findings? prolly a bucket agg query. that way we are not limited by 10000 results problem

**Review by eirsep — CHANGES_REQUESTED** [review]

File: src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java (hunk 2 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java:257]: As discussed offline, I'll address this in a separate PR.

File: src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java (hunk 1 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportListIOCsAction.java:192]: As discussed offline, I'll address this in a separate PR.

**Review by jowg-amazon — APPROVED** [review]

**Review by eirsep — APPROVED** [review]

**Pull request merged**